File: app/src/main/java/mx/dev/franco/automusictagfixer/di/ResourceManagerModule.java
package mx.dev.franco.automusictagfixer.di;

import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import mx.dev.franco.automusictagfixer.utilities.resource_manager.AndroidResourceManager;
import mx.dev.franco.automusictagfixer.utilities.resource_manager.ResourceManager;

@Module
public class ResourceManagerModule {
    @Provides
    @Singleton
    ResourceManager provideResourceManager(Context context) {
        return new AndroidResourceManager(context);
    }
}

File: app/src/main/java/mx/dev/franco/automusictagfixer/di/ResourceManagerModule.kt
package mx.dev.franco.automusictagfixer.di

import android.content.Context
import dagger.Module
import dagger.Provides
import mx.dev.franco.automusictagfixer.utilities.resource_manager.AndroidResourceManager
import mx.dev.franco.automusictagfixer.utilities.resource_manager.ResourceManager
import javax.inject.Singleton

@Module
class ResourceManagerModule {
    @Provides
    @Singleton
    fun provideResourceManager(application: Context): ResourceManager = AndroidResourceManager(application)

}